demo/docker-compose.yaml

```diff
@@ -0,0 +1,31 @@
+services:
+  tutorial:
+    image: ghcr.io/kit-mrt/arbitration_graphs_pacman_tutorial:$VERSION
+    build:
+      context: .
+      args:
+        - VERSION=$VERSION
+      target: tutorial
+    profiles: ["tutorial"]
```

Comment:
Nice, didn't know this one yet. But with the demo service not having a profile, wouldn't that just launch both services if you add `--profile tutorial`?

Reply:
Yes, so I'm bit of misusing the profile thing in order to get the demo ad in the Readme as catchy as possible :see_no_evil: 

In the proposed setup, I'd suggest not to use `--profile` at all.
Instead, 
- `docker compose up` will run the demo service (only)
- `docker compose run --rm --service-ports tutorial` will be used for the tutorial (only)

In order to avoid `docker compose up --profile tutorial` from launching the demo service as well, w'd have to assign a `demo` profile to the latter.  
But, `docker compose up` wouldn't run anything then and we'd have to advertise the demo with `docker compose up demo`

Not a big deal, rather opinionated perfectionism :nerd_face: 

What do you prefer?

Reply:
Ah, I get it now. I like the simplified version for launching the demo but maybe we should add an explanatory comment somewhere? I guess it is a bit of a hack and without knowing the context it is not very obvious why the `profiles:` setting is there.

Reply:
Alright, see 729b1fe for the docs comment.

Reply:
Damn, I just realized that this affects building the images as well:
`arbitration_graphs/demo ❯ docker compose build`
now only builds the demo service, not the tutorial – that's a bit confusing :facepalm: 

On the other hand, I guess the end users won't run into this (only relevant when changing the service definition) :shrug: